File: pro_longcomprefix.c
#include<stdio.h>
#include<string.h>

void main(){
int N,i;
scanf("%d",&N);
char name[N][100000];
for(i=0;i<N;i++){
scanf("%s",name[i]);
}
int len,M=0;
for(i=0;i<N;i++){
len=strlen(name[i]);
if(M==0){
M=len;
}
if(M<len){
M=len;
}
}

char ch,newstr[len];
for(i=0;i<len;i++){
int tempN=0;
int count=0;
char MK=NULL;
 while(tempN<N){
if(MK==0 || MK==name[tempN][i]){
MK=name[tempN][i];
count++;
}
tempN++;
 }
if(count==N){
printf("%C",MK);
}
}

}

File: pro_longcomprefix.py
N=int(input())
if(1<=N):
	A=[]
	for i in range (N):
		A.append(input())
	length=0
	tempN=N
	while(tempN>0):
		tempN=tempN-1
		if(length==0 or length>len(A[tempN])):
			length=len(A[tempN])
	temp=""		
	if(length<=100000):
		f=False
		for p in range(0,length):
			
			if(f==True):
				break
			
			#print("p"),print(p)
			tempN=N
			temp=""
			count=0
			while(tempN>0):
				tempN=tempN-1
				st=A[tempN]
				#print(str(tempN)+" "+st[p])
				if(temp=="" or temp==st[p]):
					temp=st[p]
					count=count+1
				else:
					f=True
					break
				if(count==N):
					print(st[p],end="")	
	
			
